src/lib.rs

```diff
@@ -81,6 +81,11 @@ impl IoUring<squeue::Entry, cqueue::Entry> {
     pub fn new(entries: u32) -> io::Result<Self> {
         Self::builder().build(entries)
     }
+
+    /// Create an `IoUring` instance from a pre-opened file descriptor.
+    pub unsafe fn from_fd(fd: RawFd, params: Parameters) -> io::Result<Self> {
```

Comment:
if user uses an existing fd, how does he construct Parameters?

Reply:
Good point. In my use case testing I've been a little naughty and used the internal representation with a byte transmute (relying on `io_uring_params`). Definitely _not_ the route I'd want to promote.

Given that the kernel's interface struct written in `io_uring_setup` is essentially fixed forever, would it make sense to define a public version of the parameter type with Rust mappings of the attributes? Depending on where we want to handle the unsafety of validating / trusting its attributes, the interface here could take such a new struct or `Parameters` could be publicly convertible from it.

Reply:
I don't have a good idea, but I'm willing to add a `#[repr(transparent)]` to `Parameters`, and user needs to call unsafe code to convert it.